### PR TITLE
RegEx Update for Text parser

### PR DESCRIPTION
**Pull request by Starhkz**
Includes unit test

https://user-images.githubusercontent.com/67955282/135714668-9ae730d5-a609-417d-b1e3-e1f0e4a74a60.mp4

### CODE REVIEW

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart [hunk 1]
```diff
@@ -1,61 +1,62 @@
 import 'package:flutter_parsed_text/flutter_parsed_text.dart';
 import 'package:stacked/stacked.dart';
-import 'package:stacked_services/stacked_services.dart';
 
-import '../../../../app/app.locator.dart';
-import '../../../../utilities/enums.dart';
+import 'package:url_launcher/url_launcher.dart';
```
Comment (pauleke65) [inline lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:4]: This shouldn't be here.
Use the one in lib/utilities/utilities.dart
You can fix it up there if it's not working

**Review by jeremiahseun — APPROVED** [review]
Awesome.
LGTM

**Review by pauleke65 — CHANGES_REQUESTED** [review]
I didn't mean remove the url launcher completely, i meant import it from the utilities folder instead of importing it here

**Review by YenumJay — CHANGES_REQUESTED** [review]
Good job so far, few changes and you are good.

1. Use standard practices to describe your test cases.

2. Mock your unit test (Use the current Mockito Test Setup already done and register your service and mock your values)

3. Record your device while running your unit test to show the green checks and output results.

**Review by jeremiahseun — DISMISSED** [review]
LGTM

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
-        onTap: ((_) {})),
+        pattern: r'(?<=\*)(.*)(\*)',
+        style: AppTextStyles.archiveTextStyle),
+
+// To call channels using the hashtag
     MatchText(
-        type: ParsedType.CUSTOM,
-        pattern:
-            '(---( )?(`)?spoiler(`)?( )?---)\n\n(.*?)\n( )?(---( )?(`)?spoiler(`)?( )?---)',
+        pattern: r'\B#+([\w]+)\b',
```
Comment (pauleke65) [inline lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:46]: no parsed type here, use custom

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
-    ),
-    MatchText(
-        pattern: r'\B#+([\w]+)\b',
+        pattern: 'keyword',
         style: AppTextStyles.messageTextButton,
         onTap: ((_) {})),
+    //gets '@' mentions
     MatchText(
-        pattern: r'lon',
+        pattern: r'\B@+([\w]+)\b',
```
Comment (pauleke65) [inline lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:58]: and here

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart [hunk 4]
```diff
@@ -1,61 +1,61 @@
...
-        final _bottomSheetService = locator<BottomSheetService>();
-        await _bottomSheetService.showCustomSheet(
-          variant: BottomSheetType.user,
-          isScrollControlled: true,
-        );
-      },
-    ),
-    MatchText(
-        pattern: r'\B#+([\w]+)\b',
+        pattern: 'keyword',
```
Comment (pauleke65) [inline lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:53]: here too

File: test/services/text_parser_test.dart [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    test("Returns true if it contains a Url", () async {
+      expect(
+          RegExp(r'(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})')
+              .hasMatch(sample3),
+          false);
+    });
+    test("Returns true if it contains '@'mentions", () async {
+      expect(RegExp(r'\B@+([\w]+)\b').hasMatch(sample1), true);
+    });
+    test("Returns true if it contains hashtag", () async {
```
Comment (pauleke65) [inline test/services/text_parser_test.dart:19]: hashtag or keyword?

Comment (Starhkz) [reply test/services/text_parser_test.dart]: Edited

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart (hunk 3 above)
Comment (Starhkz) [reply lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:60]: Done

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Please make requested changes and 
write an additional test for this for the channels text parser -> "this is a channel called @#channel-name-here and fine"

This should return #channel-name-here , not #channel, or @#channel

**Review by pauleke65 — APPROVED** [review]
LGTM

**Review by YenumJay — APPROVED** [review]
Great, Looks good.

**Pull request merged**